torch/nn/utils/spectral_norm.py

```diff
@@ -120,7 +120,7 @@ def apply(module: Module, name: str, n_power_iterations: int, dim: int, eps: flo
 
         fn = SpectralNorm(name, n_power_iterations, dim, eps)
         weight = module._parameters[name]
-        if isinstance(weight, torch.nn.parameter.UninitializedParameter):
+        if isinstance(weight, torch.nn.parameter.UninitializedParameter) or weight is None:
```

Comment:
That's not really the right error when `weight is None`.
For this one, we only want to say that you cannot apply spectral_norm on a Parameter that is None.

Good catch though!

Sadly, I note the inconsistency with the weight_norm that implements this slightly differently and so doesn't have type issue: https://github.com/pytorch/pytorch/blob/aa62d305de107f46e4148ed46c52f44a73212299/torch/nn/utils/weight_norm.py#L38
(you an ignore it for this PR).